### ISSUE TITLE
Drag & Drop causing "missing" tabs

### ISSUE THREAD

**Issue by harrim4n** [piroor/treestyletab#1584]
## Short description
When dragging tabs from one window to another, they go missing or cause new windows to open.

## Steps to reproduce

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Open another window (Ctrl+N)
 4. Open a new tab in one window
 5. Drag tab to other window (onto TST sidebar)

## Expected result
The tab should be moved to the other window.

## Actual result
The tab "disappears" from TST. It still shows up in the "normal" firefox tabs in the new window.
Sometimes, instead of disappearing, a third window is opened with an empty tab and the dragged tab. Not sure what the difference is (when which scenario happens), the steps to reproduce appear to be the same.

## Environment

 * Platform (OS): Arch Linux
 * Version of Firefox: 57.0 "Mozilla Firefox for Arch Linux archlinux - 1.0"
 * Version (or revision) of Tree Style Tab: TST 2.2.11

I attached a log file from a clean install while running the steps to reproduce.
During the "Array [ 4 ] 1 1" messages, nothing happened in the window the tab was dragged to. After the messages stop, the window changes the view to the "new" tab.
If the default firefox tabs are hidden via the CSS, the tab appears to "disappear" completly and is inaccesible (at least via the UI).

[drag_tab.log](https://github.com/piroor/treestyletab/files/1486023/drag_tab.log)

**Comment by dudelson** [comment]
I can confirm erratic behaviour when dragging TST tabs to new windows. In my case, the tab sometimes appears in the TST sidebar in the new window, sometimes not, but in every case a copy of the tab remains in the TST sidebar of the old window as well. However, clicking on this tab doesn't switch to it or anything, it's completely unresponsive.

OS, firefox version, and TST version are the same as @harrim4n.

**Comment by SXZ1** [comment]
Try to press F1 twice in both windows. Do missing tabs appear after that?

**Comment by Septolum** [comment]
They do for me, but the tree structure is then messed up, as in the parent child relationships have been shifted independent of the tabs

**Comment by piroor** [comment]
Sorry, there is a regression around this operation. Recent commit should fix this...

**Comment by Lej77** [comment]
@piroor Firefox had an bug with tab ids for tabs moved between windows. I noticed this about a month ago when I did some testing with web extensions for Firefox 57. I did some testing and It doesn't seem to have been fixed yet.

The bug affected only tabs that had been moved between windows. It caused tab ids that were provided by some event listeners to differ from those provided by `browser.tabs.query({});`. These incorrect tab ids could still be used to affect the tabs though so it was more like each such tab had 2 different tab ids that were both valid.

--------------------------------

From what I can remember this was how it affect things:

`browser.tabs.onActivated` was affect.
`browser.tabs.onCreated` wasn't affected.
`browser.tabs.onUpdated` wasn't affected.
`browser.tabs.onRemoved` was affected.
`browser.tabs.onMoved` was affected.
`browser.tabs.onAttached` was affected.
`browser.tabs.onDetached` was affected.

The events that was affected provided a different tab id from `browser.tabs.query({});`.

--------------------------------
